### PR TITLE
ESP32: Add MAX_FABRICS config option

### PR DESCRIPTION
**Pull request by wqx6**
#### Problem
The `CHIP_CONFIG_MAX_FABRICS` config option should be added for ESP32 platform.

#### Change overview
Add `MAX_FABRICS` config option in chip component

#### Testing
compiling success for all-clusters-app

### CODE REVIEW

File: config/esp32/components/chip/Kconfig [hunk 1]
```diff
@@ -47,6 +47,13 @@ menu "CHIP Core"
                 A Binding object is used to configure how the local device communicates with
                 a remote entity, be it a cloud service, a mobile application, or another device.
 
+        config MAX_DEVICE_ADMINS
```
Comment (bzbarsky-apple) [inline config/esp32/components/chip/Kconfig:50]: This should be called MAX_FABRICS.

File: config/esp32/components/chip/Kconfig [hunk 2]
```diff
@@ -47,6 +47,13 @@ menu "CHIP Core"
...
+            int "Max device admins"
+            range 0 16
```
Comment (bzbarsky-apple) [inline config/esp32/components/chip/Kconfig:52]: The spec-allowed range for this value is 5-255.

**Review by bzbarsky-apple — DISMISSED** [review]

**Review by woody-apple — APPROVED** [review]

File: config/esp32/components/chip/Kconfig (hunk 1 above)
Comment (wqx6) [reply config/esp32/components/chip/Kconfig]: Changed

File: config/esp32/components/chip/Kconfig (hunk 2 above)
Comment (wqx6) [reply config/esp32/components/chip/Kconfig]: Changed

**Review by woody-apple — APPROVED** [review]
Fast tracking platform changes

**Review by dhrishi — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

**Pull request merged**